File: flutter/lib/domain/use_case/flashcard/delete_flashcard_use_case.dart
import 'package:flashcard/domain/entities/flashcard.dart';

import '../../repository/flashcard_repository.dart';

class DeleteFlashcardUseCase {
  final FlashcardRepository _flashcardRepository;

  DeleteFlashcardUseCase(this._flashcardRepository);

  Future<void> call(Flashcard flashcard) async {
    await _flashcardRepository.deleteFlashcard(flashcard);
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/flashcard/DeleteFlashcardUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase.flashcard

import com.synergyboat.flashcardAi.domain.entities.Flashcard
import com.synergyboat.flashcardAi.domain.repository.FlashcardRepository

class DeleteFlashcardUseCase(
    private val repository: FlashcardRepository
) {
    suspend operator fun invoke(flashcard: Flashcard) {
        repository.deleteFlashcard(flashcard)
    }
}